app/assets/javascripts/Components/course_summaries_table.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import {render} from 'react-dom';
+
+import ReactTable from 'react-table';
+
+
+class CourseSummaryTable extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      grade_columns: [],
+    };
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+
+    $.ajax({
+      url: Routes.populate_course_summaries_path(),
+      dataType: 'json',
+    }).then(res => {
+      let grade_column = res.marks.map(c =>
+        Object.assign([],
```

Comment:
This is not a good use of `Object.assign`, and possibly contributes to the problem of "having" to use `JSON.parse` below. If you need to reformat the data a little bit, either do it in the controller or just manually return the correct Object here (i.e., the right dictionaries with good keys and values).